### DISCUSSION TITLE
How does this actually work?

### DISCUSSION THREAD

**Discussion by ChrisMiami** [intitni/CopilotForXcode#261] [General]
In VSCode, when I'm connected to CoPilot, some grey text appears at my cursor position and if I like the code in it, I press tab and blammo! New code.

The only thing I've noticed different since I installed this extension in XCode is a little grey dot above the line I'm working on, but when I try to stab it with the mouse, nothing happens (well, it goes away, but that's pretty much nothing).

When is this plugin going to suggest something to me?

**Comment by KieranConlon** [comment]
The first thing to check is that you have the setup correct - the project README has a section that describes what prerequisites are needed, enabling the extension, granting permissions and setting up the app.

Bear in mind that some features will require that you have a valid account or subscription to some 3rd-party services - again, check the prerequisites.

Use the Copilot app's `Service` tab to setup / sign-in to the supported 3rd-party services.
![image](https://github.com/intitni/CopilotForXcode/assets/33759139/8449886e-0e01-42b3-ae47-3e4b240cc2d0)

---

The commands can be invoked from Xcode's menu bar. 
For example...
Get Suggestion:  Editor / Copilot / Get Suggestions
![image](https://github.com/intitni/CopilotForXcode/assets/33759139/7b41d33a-71c5-4d0a-ab7f-0d3761fe1ef3)

In the README, look at the section on key bindings to make accessing the various functions much easier.


You may also want to get real-time code suggestions; if so, make sure this is enabled - you can do this from the settings in the Copilot app.
![image](https://github.com/intitni/CopilotForXcode/assets/33759139/7f02d777-ffda-4bcc-806d-e3f898364bff)
